internal/pkg/csumio/csumreader_test.go

```diff
@@ -184,3 +184,191 @@ func TestReader_CorrectChecksum_VerificationEnabled(t *testing.T) {
 		})
 	}
 }
+
+// TestReader_ReadFactorCoverage exhaustively validates csumio.Reader.Read.
+// The test matrix uses pairwise coverage across three factors:
+//
+//	Factor A: requested len(p) relative to chunkSize (> chunk / == chunk / < chunk).
+//	Factor B: remaining unread bytes relative to len(p) (> p / == p / < p).
+//	Factor C: offset state (aligned at chunk boundary vs. mid-chunk).
+//
+// These three factors yield nine cases (case01–case09). Each entry below sets the total payload size,
+// pre-consumes bytes when needed, and issues a single Read with len(p) = chunkSize±1 so that the
+// target call experiences the intended combination while checksum verification remains enabled.
+func TestReader_ReadFactorCoverage(t *testing.T) {
+	chunkSize := csumio.MinimumChunkSize
+
+	halfChunk := chunkSize / 2
+	extraMargin := halfChunk
+
+	readLenGreater := chunkSize + 1
+	readLenEqual := chunkSize
+	readLenSmaller := chunkSize - 1
+
+	type matrixCase struct {
+		name           string
+		description    string
+		readLen        int
+		initialConsume int
+		totalLen       int
+	}
+
+	cases := []matrixCase{
```

Comment:
You don't have to define `matrixCase` explicitly.


```suggestion
	cases := []struct {
		name           string
		description    string
		readLen        int
		initialConsume int
		totalLen       int
	}{
```